src/diebold_mariano.jl

```diff
@@ -0,0 +1,77 @@
+# diebold_mariano.jl
+# Diebold-Mariano
+#
+# Copyright (C) 2020   Guilherme Bodin
+#
+# Permission is hereby granted, free of charge, to any person obtaining
+# a copy of this software and associated documentation files (the
+# "Software"), to deal in the Software without restriction, including
+# without limitation the rights to use, copy, modify, merge, publish,
+# distribute, sublicense, and/or sell copies of the Software, and to
+# permit persons to whom the Software is furnished to do so, subject to
+# the following conditions:
+#
+# The above copyright notice and this permission notice shall be
+# included in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+# LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+# OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+# WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+export DieboldMarianoTest
+
+struct DieboldMarianoTest <: TTest
+    n::Integer     # number of observations
+    df::Integer    # degrees of freedom
+    t::Real        # test statistic
+end
+
+"""
+    DieboldMarianoTest(x::AbstractVector{<:Real}, y::AbstractVector{<:Real})
+
+Perform the modified Diebold-Mariano test proposed by Harvey, Leybourne and Newbold of the null 
+hypothesis that the two methods have the same forecast accuracy. `loss` is the loss function described
+in Diebold, F.X. and Mariano, R.S. (1995) Comparing predictive accuracy. Journal of Business and 
+Economic Statistics, 13, 253-263. and `h` is the number of steps ahead of the forecast.
```

Comment:
Might also be worthwhile to give `h` a more descriptive name, e.g. `lookahead`.

Reply:
I chose `h` because is consistent with the paper.

Reply:
Sure, but the name doesn't mean anything to someone looking for the functionality who hasn't read the paper. Non-descriptive names are fine for positional arguments, but IMO one of the reasons to use keyword arguments is to be able to supply values along with descriptive names.